kafka-connect-ftp/src/main/scala/com/datamountaineer/streamreactor/connect/ftp/source/FtpMonitor.scala

```diff
@@ -124,24 +200,56 @@ class FtpMonitor(settings:FtpMonitorSettings, fileConverter: FileConverter) exte
         logger.info(s"dump entire ${current.meta.attribs.path}")
         (current.meta.inspectedNow().modifiedNow(), FileBody(current.body, 0))
     }
+  }
 
 
+  def bySlicesHandleFetchedFile(mode: MonitorMode, prevFetch: Option[FileMetaData], current: FetchedFile): (FileMetaData, FileBody) = {
+    prevFetch match {
+      case Some(previous) if mode == MonitorMode.Update && (previous.attribs.timestamp != current.meta.attribs.timestamp || previous.attribs.size != current.meta.attribs.size) =>
```

Comment:
All these case classes on `FileMetaData` objects are getting messing. 

My suggestion: add some methods to `FileMetaData` with what you want to check. 

For example if you add a `hasChangedSince` method, you could do:
```scala
case Some(previous) if mode == MonitorMode.Update && current.hasChangedSince(previous)
```

Reply:
good idea